serialization/serialization_it_test.go

```diff
@@ -86,3 +86,113 @@ func (p portableFactory) Create(classID int32) serialization.Portable {
 func (p portableFactory) FactoryID() int32 {
 	return 1
 }
+
+type StubPortable1 struct {
```

Comment:
Could you make sure that the tests and types you've created for tests match the one to the reference implementation: https://github.com/hazelcast/hazelcast/pull/16831/files

Reply:
Ported java tests in most meaningful way [3febce0](https://github.com/hazelcast/hazelcast-go-client/pull/758/commits/3febce0a1eb6828dacfd1a24f9c9ab44e65245a9).